### PR TITLE
Address resolution

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
This implements in FlyteRemote a quick feature to help the fetching of inputs and outputs and decks from a `flyte://` uri.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Other PRs:
https://github.com/flyteorg/flyteadmin/pull/546
https://github.com/flyteorg/flyteidl/pull/391

## Tracking Issue
No issues, this is just a use-ability enhancement we found

### CODE REVIEW

File: flytekit/clients/friendly.py [hunk 1]
```diff
@@ -1018,3 +1018,9 @@ def get_download_signed_url(
                 expires_in=expires_in_pb,
             )
         )
+
+    def get_data(self, flyte_url: str) -> _data_proxy_pb2.GetDataResponse:
```
Comment (kumare3) [inline flytekit/clients/friendly.py:1022]:
```suggestion
    def get_data(self, flyte_uri: str) -> _data_proxy_pb2.GetDataResponse:
```

File: flytekit/remote/remote.py [hunk 2]
```diff
@@ -221,6 +221,34 @@ def file_access(self) -> FileAccessProvider:
...
+            if data_response.HasField("literal_map"):
+                lm = LiteralMap.from_flyte_idl(data_response.literal_map)
+                return LiteralsResolver(lm.literals)
+            elif data_response.HasField("flyte_deck_download_link"):
+                if len(data_response.flyte_deck_download_link.signed_url) == 0:
+                    raise ValueError(f"Flyte url {flyte_url} resolved to empty download link")
+                d = data_response.flyte_deck_download_link.signed_url[0]
+                remote_logger.debug(f"Attempting to download {d} resolved from flyte url {flyte_url}")
+                fs = ctx.file_access.get_filesystem_for_path(d)
+                with fs.open(d, "rb") as r:
```
Comment (kumare3) [inline flytekit/remote/remote.py:240]: how about we return a open file object instead of writing to local disk?

Comment (kumare3) [reply flytekit/remote/remote.py:240]: I would love if we could put this block in here
```python
    raw_html = fs.read()
    if not ignore_jupyter and _ipython_check():
        from IPython.core.display import HTML

        return HTML(raw_html)
    return raw_html
```
https://ipython.readthedocs.io/en/stable/api/generated/IPython.display.html

I do not like writing to files. Let the users write it to file if they want

This is implemented already in flytekit and is very useful
https://github.com/flyteorg/flytekit/blob/8e5bf6caaf7f2facee8eb24a67d6a5d93d9c7019/flytekit/deck/deck.py#L93
returned here
https://github.com/flyteorg/flytekit/blob/8c05797feeaf0e47a56075bf1c725c8284157660/flytekit/core/context_manager.py#L639


maybe move this to utils
https://github.com/flyteorg/flytekit/blob/8e5bf6caaf7f2facee8eb24a67d6a5d93d9c7019/flytekit/deck/deck.py#L77

Comment (wild-endeavor) [reply flytekit/remote/remote.py]: done

**Review by pingsutw — DISMISSED** [review]

**Review by pingsutw — APPROVED** [review]

**Pull request merged**